main/exercice/fill_blanks.class.php

```diff
@@ -148,40 +286,112 @@ function updateBlanks() {
+        <p>I use a [computer] and a [pen].</p>::100,50:100,50@1
+                                              ++++++++-------**
+                                                --- -- --- -- -
+                                                  A B  (C) (D)(E)
+        +++++++ : required, weighting of each words
+        ------- : optional, input width to display, 200 if not present
+        ** : equal @1 if "Allow answers order switches" has been checked, @ otherwise
+        A : weighting for the word [computer]
+        B : weighting for the word [pen]
+        C : input width for the word [computer]
+        D : input width for the word [pen]
+        E : equal @1 if "Allow answers order switches" has been checked, @ otherwise
+        */
+		if($nb > 0) {
+			$answer .= '::';
+            // weighting
+			for($i=0 ; $i < $nb ; ++$i) {
+                // enter the weighting of word $i
+				$answer .= $form->getSubmitValue('weighting['.$i.']');
+                // not the last word, add ","
+                if ($i != $nb - 1) {
+                    $answer .= ",";
+                }
+                // calculate the global weightning for the question
+				$this -> weighting += $form->getSubmitValue('weighting['.$i.']');
+			}
 
-		if ($nb > 0) {
-		  	$answer .= '::';
-			for ($i=0 ; $i < $nb; ++$i) {
-                $blankItem = $blanks[0][$i];
-                $replace = array("[", "]");
-                $newBlankItem = str_replace($replace, "", $blankItem);
-                $newBlankItem = "[".trim($newBlankItem)."]";
-                $answer = str_replace($blankItem, $newBlankItem, $answer);
-                $answer .= $form->getSubmitValue('weighting['.$i.']').',';
-                $this -> weighting += $form->getSubmitValue('weighting['.$i.']');
+            // input width
+            $answer .= ":";
+            for($i=0 ; $i < $nb ; ++$i) {
+                // enter the width of input for word $i
+                $answer .= $form->getSubmitValue('sizeofinput['.$i.']');
+                // not the last word, add ","
+                if ($i != $nb - 1) {
+                    $answer .= ",";
+                }
             }
-            $answer = api_substr($answer, 0, -1);
 		}
 
-		$is_multiple = $form->getSubmitValue('multiple_answer');
-		$answer.= '@'.$is_multiple;
-
-		$this->save();
+        // write the blank separator code number
+        // see function getAllowedSeparator
+        /*
+            0 [...]
+            1 {...}
+            2 (...)
+            3 *...*
+            4 #...#
+            5 %...%
+            6 $...$
+         */
+        $answer .= ":".$form->getSubmitValue('select_separator');
+
+        // Allow answers order switches
+		$is_multiple = $form -> getSubmitValue('multiple_answer');
+		$answer.='@'.$is_multiple;
+
+		$this -> save();
         $objAnswer = new answer($this->id);
-        $objAnswer->createAnswer($answer, 0, '', 0, '1');
+        $objAnswer->createAnswer($answer,0,'',0,1);
```

Comment:
All parameters should have a space after the comma (,) and before the parameter, like it was before your change, sorry. This is a coding convention now.